azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java

```diff
@@ -45,128 +47,144 @@ public Account account() throws AzureToolkitAuthenticationException {
     }
 
     public List<Account> accounts() {
-        return buildAccountMap(null).values().stream().filter(Account::checkAvailable).collect(Collectors.toList());
+        List<Account> accountList = Flux.fromIterable(buildAccountMap().values()).map(Supplier::get).collectList().block();
+        return accountList.stream().filter(Account::checkAvailable).collect(Collectors.toList());
     }
 
-    public AzureAccount login(@Nonnull Account targetAccount) throws LoginFailureException {
-        account = targetAccount;
-        if (account.checkAvailable()) {
-            if (!account.isAuthenticated()) {
-                account.authenticate().block();
-            }
-        } else {
-            if (account.entity.getLastError() != null) {
-                throw new LoginFailureException(account.entity.getLastError().getMessage(), account.entity.getLastError());
-            } else {
-                throw new LoginFailureException(String.format("Cannot get credential from auth method '%s'.", targetAccount.getMethod()));
-            }
-        }
-        return this;
+    public Account account(AuthType type) {
+        Account ac = buildAccountMap().get(type).get();
+        ac.checkAvailable();
```

Comment:
will we throw exception in `checkAvailable`? If so, `accounts()` may be break once there are inliad account, if not, why we call `ac.checkAvailable()` here?

Reply:
checkAvailable will not, according to new discussion, we will make checkAvailable return Mono<void>

Reply:
will remove checkAvailable in future PR